src/lib.rs

```diff
@@ -405,4 +479,80 @@ mod tests {
         }
         assert_eq!(iter.next(), None);
     }
+
+    #[test]
+    fn test_delta() {
+        // The trickiness here almost entirely relates to negative maximum values for signed
+        // integer types
+        assert_eq!(delta::<u8, u8>(0, 2), 2);
+        assert_eq!(delta::<u8, u8>(0, u8::max_value()), u8::max_value());
+        assert_eq!(delta::<i8, u8>(-2, 0), 2);
+        assert_eq!(delta::<i8, u8>(-2, 2), 4);
+        assert_eq!(delta::<i8, u8>(i8::min_value(), 0), 128);
+        assert_eq!(delta::<i8, u8>(0, i8::max_value()), 127);
+        assert_eq!(
+            delta::<i8, u8>(i8::min_value(), i8::max_value()),
+            u8::max_value()
+        );
+        assert_eq!(delta::<i8, u8>(i8::min_value(), i8::min_value()), 0);
+        assert_eq!(
+            delta::<i32, u32>(i32::min_value(), i32::max_value()),
+            (i32::max_value() as u32) * 2 + 1
+        );
+        assert_eq!(
+            delta::<i32, usize>(i32::min_value(), i32::max_value()),
+            (i32::max_value() as usize) * 2 + 1
+        );
+    }
+
+    #[test]
+    fn test_inv_delta() {
+        // These tests are, in essence, those from delta with the last two values swapped (i.e.
+        // assert_eq!(delta(x, y), z) becomes assert_eq!(inv_delta(x, z), y).
+        assert_eq!(inv_delta::<u8, u8>(0, 2), 2);
+        assert_eq!(inv_delta::<u8, u8>(0, u8::max_value()), u8::max_value());
+        assert_eq!(inv_delta::<i8, u8>(-2, 2), 0);
+        assert_eq!(inv_delta::<i8, u8>(-2, 4), 2);
+        assert_eq!(inv_delta::<i8, u8>(i8::min_value(), 128), 0);
+        assert_eq!(inv_delta::<i8, u8>(0, 127), i8::max_value());
+        assert_eq!(
+            inv_delta::<i8, u8>(i8::min_value(), u8::max_value()),
+            i8::max_value()
+        );
+        assert_eq!(inv_delta::<i8, u8>(i8::min_value(), 0), i8::min_value());
+        assert_eq!(
+            inv_delta::<i32, u32>(i32::min_value(), ((i32::max_value() as u32) * 2 + 1).as_()),
+            i32::max_value()
+        );
+        assert_eq!(
+            inv_delta::<i32, usize>(
+                i32::min_value(),
+                ((i32::max_value() as usize) * 2 + 1).as_()
+            ),
+            i32::max_value()
+        );
+    }
+
+    #[test]
+    fn efficient_range() {
+        let pv = PackedVec::new(vec![9998, 9999, 10000]);
+        // lot less than the 14 bits we'd need without our range optimisation.
```

Comment:
lot less -> a lot less

Reply:
Also, this *is* a great improvement!

Reply:
Oops, that comment got chopped off somewhere :/

Reply:
Actually, I'm not sure the comment is useful anymore...